### ISSUE TITLE
setup.cfg's "version = attr: jinja2.__version__" causes issues when installing from source

### ISSUE THREAD

**Issue by Vultaire** [pallets/jinja#1610]
When installing jinja2 from sources in a fresh virtualenv, setup.py breaks unless markupsafe is installed prior.  I seem to have traced it to the ```version = attr: jinja2.__version__``` line in setup.cfg; if I replace that with a hard-coded version (e.g. 3.0.3), then installation works.

This is problematic since some tools install files via requirements.txt using the ```--no-binary :all:```, and since setup.cfg pulls code from jinja which in turn tries to import markupsafe before markupsafe can be installed as a dependency, things break.

How to reproduce the bug:
* Prerequisite: Create and activate a fresh virtualenv via ```python3 -m venv venv; . venv/bin/activate```.  (Ensure the venv doesn't already exist, of course.)
* Method 1: Download and unzip the jinja2 tarball, cd inside, then run ```python setup.py install```.
* Method 2: Install via ```pip install --upgrade --no-binary :all: 'jinja2>=3.0.3'```

Full, complete example and traceback showing the error:

```
paul-goins@dell2018:~$ rm -rf venv
paul-goins@dell2018:~$ python3 -m venv venv
paul-goins@dell2018:~$ . venv/bin/activate
(venv) paul-goins@dell2018:~$ pip install --upgrade --no-binary :all: 'jinja2>=3.0.3'
Collecting jinja2>=3.0.3
  Using cached Jinja2-3.0.3.tar.gz (269 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/paul-goins/venv/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-7wo6yg5s/jinja2/setup.py'"'"'; __file__='"'"'/tmp/pip-install-7wo6yg5s/jinja2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-7wo6yg5s/jinja2/pip-egg-info
         cwd: /tmp/pip-install-7wo6yg5s/jinja2/
    Complete output (35 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-7wo6yg5s/jinja2/setup.py", line 4, in <module>
        setup(
      File "/home/paul-goins/venv/lib/python3.8/site-packages/setuptools/__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib/python3.8/distutils/core.py", line 121, in setup
        dist.parse_config_files()
      File "/home/paul-goins/venv/lib/python3.8/site-packages/setuptools/dist.py", line 701, in parse_config_files
        parse_configuration(self, self.command_options,
      File "/home/paul-goins/venv/lib/python3.8/site-packages/setuptools/config.py", line 121, in parse_configuration
        meta.parse()
      File "/home/paul-goins/venv/lib/python3.8/site-packages/setuptools/config.py", line 426, in parse
        section_parser_method(section_options)
      File "/home/paul-goins/venv/lib/python3.8/site-packages/setuptools/config.py", line 399, in parse_section
        self[name] = value
      File "/home/paul-goins/venv/lib/python3.8/site-packages/setuptools/config.py", line 184, in __setitem__
        value = parser(value)
      File "/home/paul-goins/venv/lib/python3.8/site-packages/setuptools/config.py", line 515, in _parse_version
        version = self._parse_attr(value, self.package_dir)
      File "/home/paul-goins/venv/lib/python3.8/site-packages/setuptools/config.py", line 349, in _parse_attr
        module = import_module(module_name)
      File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
      File "<frozen importlib._bootstrap>", line 991, in _find_and_load
      File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
      File "<frozen importlib._bootstrap_external>", line 848, in exec_module
      File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
      File "/tmp/pip-install-7wo6yg5s/jinja2/src/jinja2/__init__.py", line 8, in <module>
        from .environment import Environment as Environment
      File "/tmp/pip-install-7wo6yg5s/jinja2/src/jinja2/environment.py", line 15, in <module>
        from markupsafe import Markup
    ModuleNotFoundError: No module named 'markupsafe'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

Environment:

- Python version: 3.8.10
- Jinja version: 3.0.3

Best Regards,
Paul Goins

**Comment by davidism** [comment]
You are using an very, very old, unsupported version of pip and/or setuptools. Use `pip install -U pip setuptools wheel` to update the common set of build tools. Duplicate of pallets/markupsafe#285

**Comment by Vultaire** [comment]
Thanks - I had tried with a new pip, but not a new setuptools.  Good to know!
